framework/src/main/java/org/tron/core/net/peer/PeerConnection.java

```diff
@@ -167,7 +167,7 @@ public void setBlockBothHave(BlockId blockId) {
   }
 
   public boolean isIdle() {
-    return advInvRequest.isEmpty() && syncBlockRequested.isEmpty() && syncChainRequested == null;
```

Comment:
What were the considerations of this previous judgment?

Reply:
This logic has been around for a long time, and this special scenario may not have been considered before.